lfc-scheduler/cmd/scheduler/main.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/keptn-sandbox/lifecycle-controller/lfc-scheduler/pkg/klcpermit"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/component-base/cli"
+	"k8s.io/kubernetes/cmd/kube-scheduler/app"
+	"os"
+	"time"
+)
+
+func main() {
+
+	rand.Seed(time.Now().UnixNano())
+	command := app.NewSchedulerCommand(
+		app.WithPlugin(klcpermit.Name, klcpermit.New),
```

Comment:
Do you think it would make sense to hook in an earlier stage? (https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/)

Reply:
In my mind pre-deployment checks could take a while and it would not make sense to not discover in the meantime that there are other reasons for the pod not to be scheduled e.g. filters make it impossible to find a compatible node. We discussed having prechecks and scheduling steps in parallel up to resource allocation, but maybe also pre-Score could be an option, just I could not think of a reason why we would like to interfere with the node choice. What do you think could be another approach?

Reply:
Good point, think it makes sense to leave this in the permit-phase at the moment :-)